### PR TITLE
Build self-extracting-executable utils

### PR DESCRIPTION
**Pull request by yakov-olkhovskiy**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

We need to build utils/self-extracting-executable whenever we want to build compressed binary.
Required for #38653

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]
LGTM

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,6 +13,16 @@ else()
     set(RECONFIGURE_MESSAGE_LEVEL WARNING)
 endif()
 
+# Setup NATIVE_BUILD_TARGETS list for further native build
+# when cross-compiling and host architecture differs from target architecture.
```
Comment (rschu1ze) [inline CMakeLists.txt:17]: The comment could give a bit more context, e.g.

```
With cross-compiling, all targets are build for the target platform which usually different from the host 
platform. This is problematic if a build artifact X (e.g. a file or an executable) is generated by running 
another executable Y previously produced in the build. This is solved by compiling and running Y for/on 
the host platform. Add target to the list:
   add_native_target(<target> ...)
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -568,3 +578,36 @@ add_subdirectory (tests)
...
+
+    execute_process(
+        COMMAND ${CMAKE_COMMAND}
+            "-G${CMAKE_GENERATOR}"
+            "-DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}"
+            "-DCMAKE_MAKE_PROGRAM=${CMAKE_MAKE_PROGRAM}"
+            "-DCMAKE_C_COMPILER=${CMAKE_C_COMPILER}"
+            "-DCMAKE_CXX_COMPILER=${CMAKE_CXX_COMPILER}"
+        ${CMAKE_SOURCE_DIR}
+        WORKING_DIRECTORY "${NATIVE_BUILD_DIR}"
```
Comment (rschu1ze) [inline CMakeLists.txt:606]: I never had to specify the generator (l. 600/602) manually, and afaik we anyways only support Ninja. The generator is automatically set by "preload.cmake". Could we remove l. 600/602?

File: utils/self-extracting-executable/compressor.cpp [hunk 3]
```diff
@@ -9,17 +9,17 @@
...
-
-// dependencies
-#include <machine/endian.h>
-#include <libkern/OSByteOrder.h>
 
-// define 64 bit macros
-#define htole64(x) OSSwapHostToLittleInt64(x)
+#if (defined(OS_DARWIN) || defined(OS_FREEBSD)) && defined(__GNUC__)
+#   include <machine/endian.h>
+#elif !defined(_MSC_VER)
```
Comment (rschu1ze) [inline utils/self-extracting-executable/compressor.cpp:15]: Guess that can be just `#else` ... we are not using the MS VC compiler.

**Review by rschu1ze — APPROVED** [review]

**Pull request merged**